main.py

```diff
@@ -107,6 +107,21 @@ def update(self, name: str, serial: str, message: object) -> None:
       self.heat_target.labels(name=name, serial=serial).set(heat_target)
       self.quality_target.labels(name=name, serial=serial).set(message.quality_target)
       self.filter_life.labels(name=name, serial=serial).set(filter_life)
+    elif isinstance(message, dyson_pure_state.DysonPureCoolState):
```

Comment:
What do you think about combining the elif's on line 88 and 110?

DysonPureHotCoolState inherits from DysonPureCoolState, so something like this should work and reduce code duplication:

```
elif isinstance(message, dyson_pure_state.DysonPureCoolState):
  self.fan_mode.labels(...)
  (rest of the common PureCool code)

  if isinstance(message, dyson_pure_state.DysonPureHotCoolState):
    self.heat_mode.labels(...)
    (rest of the heat code)
```

Reply:
Yes, thank you for the suggestion! I've updated the PR. Looks much better now.